api/app/googleflood.py

```diff
@@ -30,13 +38,31 @@ class InundationMapSet(BaseModel):
     inundationMaps: List[InundationMap]
 
 
+# This cache helps avoid making repeated requests to the Google Floods API
+CACHE_TIMEOUT = 600  # 10 minutes
 GOOGLE_FLOODS_API_KEY = os.getenv("GOOGLE_FLOODS_API_KEY", "")
 if GOOGLE_FLOODS_API_KEY == "":
     logger.warning("Missing backend parameter: GOOGLE_FLOODS_API_KEY")
 
 
 def make_google_floods_request(url, method="get", data=None, retries=1, timeout=10):
-    """Make a request with retries and error handling."""
+    """Make a request with retries and error handling, caching the result for 5 minutes."""
+    # Create a unique cache key based on the request parameters
+    request_data = json.dumps(data, sort_keys=True) if data else ""
+    cache_key = f"{url}_{method}_{request_data}"
+    cache_filepath = get_cached_filepath(
+        prefix="google_floods", data=cache_key, extension="json"
+    )
+
+    # Check if the cached file is valid and not expired
+    if is_file_valid(cache_filepath):
+        cached_data = get_json_file(cache_filepath)
+        cache_age = get_cache_age(cache_filepath)
+        if cache_age < CACHE_TIMEOUT:
+            logger.debug("Returning cached data.")
+            return cached_data
```

Comment:
You could return `get_json_file(cache_filepath)` on the last line and remove the step above `cache_age = ` ... (i.e. remove L59)

Reply:
Updated